plugins/inputs/solrmetrics/solrmetrics.go

```diff
@@ -0,0 +1,413 @@
+			subMetsMap, ok := v.(map[string]interface{})
+			if !ok {
+				acc.AddError(fmt.Errorf("error of converting map"))
+				continue
+			}
+			subMets := parseMetrics(subMetsMap, acc)
+			if len(subMets) == 0 {
+				continue
+			}
+			for km, vm := range subMets {
+				key := k + delimeter + km
+				f[key] = vm
+			}
+		default:
+			acc.AddError(fmt.Errorf("error of parsing, unknow type. %s=%s", k, v))
+			continue
+		}
+	}
+	return f
+}
+
+// getMetricTags parces metric name to collection, shard, replica, metric
+func getCollectionTags(metricName string) ([]string, error) {
+	ct := strings.SplitN(metricName, ".", 4)
+	if len(ct) < 4 {
+		return nil, fmt.Errorf("wrong metric name %s", metricName)
+	}
+	return ct, nil
+}
+
+func metricsToSeries(metrics Metrics, addTags Tags, acc telegraf.Accumulator) (res []Serie) {
+	mTime := time.Now()
+	series := make(map[string]Serie)
+
+	// Create series for standard groups
+	for _, group := range []string{"jvm", "jetty", "node"} {
+		series[group] = Serie{mesName: group, fields: make(Fields), tags: make(Tags), timestamp: mTime}
+		copyStrMaps(addTags, series[group].tags)
+	}
+
+	// Parce metrics by a group
+	for k, v := range metrics {
+		// standard groups
+		for _, group := range []string{"jvm", "jetty", "node"} {
+			if strings.Contains(k, "solr"+delimeter+group) {
+				newSerie := strings.Replace(k, "solr."+group+":", "", 1)
+				series[group].fields[newSerie] = v
+			}
+		}
+		// Core group
+		// Each core of any collection should be present as a separate serie with uniq tags
+		if strings.Contains(k, "solr.core") {
+			newSerie := strings.Replace(k, "solr.core"+delimeter, "", 1)
+			// Decompose a name of a metric
+			colTags, err := getCollectionTags(newSerie)
+			if err != nil || len(colTags) != 4 {
+				acc.AddError(err)
+				continue
+			}
+			collection, shard, replica, metricName := colTags[0], colTags[1], colTags[2], colTags[3]
+			newSerie = collection + shard + replica
+			// Check and create an unique serie if it doesn't exist
+			if _, ok := series[newSerie]; !ok {
+				series[newSerie] = Serie{mesName: "core", fields: make(Fields), tags: make(Tags), timestamp: mTime}
+				// Add requeremnted tags
+				copyStrMaps(addTags, series[newSerie].tags)
+				series[newSerie].tags["collection"] = collection
+				series[newSerie].tags["shard"] = shard
+				series[newSerie].tags["replica"] = replica
+			}
+			series[newSerie].fields[metricName] = v
+		}
+	}
+	res = []Serie{}
+	for _, v := range series {
+		// Add a serie, if fileds contain metrics
+		if len(v.fields) > 0 {
+			res = append(res, v)
+		}
+	}
+	return res
+}
+
+// SampleConfig returns sample configuration for this plugin.
+func (s *Solr) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns the plugin description.
+func (s *Solr) Description() string {
+	return "Read stats from Solr servers and cores using Metrics API"
+}
+
+// NewSolr return a new instance of Solr
+func NewSolr() *Solr {
+	return &Solr{}
+}
```

Comment:
I think you should not add a function for this. Instantiate the struct directly.